### PR TITLE
Include e2e tests for `guaccollect`, `guacingest`, and ent

### PR DESCRIPTION
**Pull request by nathannaveen**
# Description of the PR

* Fixes https://github.com/guacsec/guac/issues/1953
* Cherry picked code from https://github.com/guacsec/guac/pull/1989

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If GraphQL schema is changed, GraphQL client updates/additions have been made
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: internal/testing/e2e/e2e [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+﻿#!/opt/homebrew/opt/bash/bin/bash
```
Comment (funnelfiasco) [inline internal/testing/e2e/e2e:1]: This should probably be `#!/bin/env bash` to be OS-agnostic.

Comment (nathannaveen) [reply internal/testing/e2e/e2e]: Thanks! I made that change for testing and forgot to revert it.

File: internal/testing/e2e/e2e [hunk 2]
```diff
@@ -17,13 +17,22 @@
...
+
+# Create the log directory if it doesn't exist
+mkdir -p "$LOG_DIR"
+
+echo "Script started" | tee -a "$LOG_FILE"
 
 guac_data_hash="254d737ebc42ca32f95033e93370f5427e3fce64"
 
-echo @@@@ Installing gql Python package
-pip install gql[all]
```
Comment (pxp928) [inline internal/testing/e2e/e2e:26]: keep this the same if its causing errors?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -136,7 +150,30 @@ jobs:
     - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d
       with:
         python-version: '3.10'
-    - run: ./internal/testing/e2e/e2e
+    - name: Print Bash version
+      run: bash --version
+    - name: Install dependencies
+      run: |
+        go mod download
+        pip install gql[all]
```
[the remark this answers is not in the document]
Comment (pxp928) [reply .github/workflows/ci.yaml:158]: keep it as part of the e2e script?

File: internal/testing/e2e/e2e (hunk 2 above)
Comment (nathannaveen) [reply internal/testing/e2e/e2e]: Sorry, used this for testing, was cleaning it up as you were giving the comment. Thanks!

File: .github/workflows/ci.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathannaveen) [reply .github/workflows/ci.yaml]: Removed, thanks

**Comment by pxp928** [comment]
@nathannaveen this can be rebased now!

**Comment by nathannaveen** [comment]
Thanks @pxp928, rebased

**Comment by pxp928** [comment]
This is ready for review now right @nathannaveen ?

**Comment by nathannaveen** [comment]
I cleaned up the code a bit, now it should be ready to review.

**Review by pxp928 — APPROVED** [review]
LGTM! small nit

**Comment by pxp928** [comment]
Great improvement to the e2e test!

**Pull request merged**